pkg/model/resource/options_test.go

```diff
@@ -0,0 +1,96 @@
+package resource_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	. "sigs.k8s.io/kubebuilder/pkg/model/resource"
```

Comment:
We normally don't use dot import for packages besides ginkgo and gomega.

Reply:
Self imports are also used with dot imports. It just needs to import its own package because it belongs to the secial resource_test package to avoid circular imports.

Reply:
This seems to be fine according to https://www.agiratech.com/12-best-golang-agile-practices-we-must-follow/#Import_Dot